storage/remote/queue_manager.go

```diff
@@ -1198,48 +1196,24 @@ func (s *shards) runShard(ctx context.Context, shardID int, queue *queue) {
 			}
 			s.mtx.Unlock()
 			if len(batch) > 0 {
-				nPendingSamples, nPendingExemplars := s.populateTimeSeries(batch, pendingData)
```

Comment:
Hm, those pending samples and pending exemplars changes are not related to marshalling are they?

1. Are we sure we don't need this functionality anymore?
2. Are we sure when the removal this functionality is NOT what is faster/more efficient vs marshalling changes?

Reply:
Yes it is related.  The code now generates protobuf messages directly from Prometheus data structures, and does not convert to protobuf structs.  The function `populateTimeSeries` used to do this conversion, and has been removed.

You're right that this is a large part of making it faster.  This is stated in the PR description and in the git commit message.